include/aspect/postprocess/particle_distribution_statistics.h

```diff
@@ -0,0 +1,144 @@
+         * point-density-function is defined on a regular grid throughout the cell.
+         */
+        bool KDE_per_particle;
+
+        /**
+         * The `granularity` variable determines the number of points at which
+         * the point-density function is defined within each cell. For example,
+         * a value of 2 means that the point-density function is defined at
+         * $2\times 2=4$ points in 2D. This variable only applies if
+         * `KDE_per_particle` is false.
+         */
+        unsigned int granularity;
+
+        /**
+         * The `bandwidth` variable scales the point-density function.
+         * Choosing an appropriate bandwidth is important because a
+         * bandwidth value which is too low or too high can result
+         * in oversmoothing or undersmoothing of the point-density function.
+         * Oversmoothing or undersmoothing results in a function which
+         *  represents the underlying data less accurately.
+         */
+        double bandwidth;
+
+        /**
+         * The `KernelFunctions` enum class is a data structure which
+         * contains the kernel functions available for use in the Kernel
+         * Density Estimator.
+         */
+        enum class KernelFunctions
+        {
+          GAUSSIAN,
+          TRIANGULAR,
+          UNIFORM,
+          CUTOFF_FUNCTION_W1_DEALII
+        };
+
+        /**
+         * `kernel_function` is an internal variable to keep track of which
+         * kernel function was read from the .prm file.
+         */
+        KernelFunctions kernel_function;
+
+        /**
+         * Fills the supplied PDF instance with values from the particles in the given cell.
+         * @param cell The cell whose particles to populate the PDF with.
+         * @param pdf The instance of ParticlePDF to fill with data from the cell's particles.
+         */
+        void fill_PDF_from_cell(const typename Triangulation<dim>::active_cell_iterator &cell, ParticlePDF<dim> &pdf);
+
+        /**
+         * This function is only called from `fill_PDF_from_cell`.
+         * It iterates through every particle in the cell and
+         * sums the value of the kernel function between the
+         * reference point and the position of the cell.
+         * @param cell The cell whose particles to populate the PDF with.
+         * @param pdf The instance of ParticlePDF to fill with data from the cell's particles.
+         * @param reference_point The point from which to get the value of the kernel function.
+         * @param table_point The index in the PDF to insert the data into.
+         * @param particles_in_cell The number of particles in the cell.
+         */
+        void fill_PDF_point_from_cell(const typename Triangulation<dim>::active_cell_iterator &cell, ParticlePDF<dim> &pdf,const Point<dim> reference_point,Point<dim, unsigned int> table_point,const unsigned int particles_in_cell);
```

Comment:
It is perhaps a bit confusing that these two functions have the same name, and that one calls the other. Maybe give the internal (second) function a name that distinguishes what it does from the first?